library/src/rocblas_auxiliary.cpp

```diff
@@ -60,56 +58,52 @@ extern "C" rocblas_status rocblas_set_pointer_mode(rocblas_handle handle, rocbla
  * ! \brief create rocblas handle called before any rocblas library routines
  ******************************************************************************/
 extern "C" rocblas_status rocblas_create_handle(rocblas_handle* handle)
+try
 {
-    // if handle not valid
     if(!handle)
         return rocblas_status_invalid_handle;
 
     // allocate on heap
-    try
-    {
-        *handle = new _rocblas_handle();
+    *handle = new _rocblas_handle;
+
+    if((*handle)->layer_mode & rocblas_layer_mode_log_trace)
+        log_trace(*handle, "rocblas_create_handle");
 
-        if((*handle)->layer_mode & rocblas_layer_mode_log_trace)
-            log_trace(*handle, "rocblas_create_handle");
-    }
-    catch(...)
-    {
-        return rocblas_status_internal_error;
-    }
     return rocblas_status_success;
 }
+catch(...)
+{
+    return rocblas_status_memory_error;
```

Comment:
Are you sure that that it should be a rocblas_status_memory_error rather than rocblas_status_internal_error?

Reply:
It almost always catches a `std::bad_alloc` from the `new` when it receives an exception, so the memory error is more descriptive than internal error. I could catch        `std::bad_alloc` and flag it as memory error, while making other exceptions return internal error.

Reply:
Normally I think that would be good enough but for this case I think  two catch blocks might be better to discern the two failure modes.  Or two try catches.   I think we may want another error enum but not for this case, discuss later today.

Reply:
See #863 for a general solution to the exception issue. If you need to catch different exceptions and make them return a different `rocblas_status`, then change `library/src/include/utility.h`, which has a universal exception -> `rocblas_status` conversion function.